### PR TITLE
Bug 2033098: Conjunction in the ProgressiveListFooter is now translatable

### PR DESCRIPTION
**Pull request by gruselhaus**
**Fixes**: 
[BZ2033098](https://issues.redhat.com/browse/OCPBUGSM-38517)

**Analysis / Root cause**: 
In https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/progressive-list/ProgressiveListFooter.tsx#L22-L29 we use an approach that does not allow the rendered text to be translated.

**Expected results**: 
The text should be translatable. We should use the internationalization API provided by the browser. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat.

**Screen shots / Gifs for design review**: 
## English
english

## Mandarin
chinese

## Korean
korean

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

**Review by jerolimov — CHANGES_REQUESTED** [review]
Unfortunately, this breaks the option to click on one of the adv. list items. Looks like a rerender loop, can you please check or ping me if we should take a look together.

**Comment by gruselhaus** [comment]
Thanks @jerolimov 🎉 I'll take a deeper look into this 👍

File: frontend/packages/dev-console/src/components/import/advanced/AdvancedSection.tsx [hunk 1]
```diff
@@ -15,58 +15,68 @@ import RouteSection from './RouteSection';
...
+
+const AdvancedSection: React.FC<AdvancedSectionProps> = ({ values, appResources }) => {
+  const { t } = useTranslation();
+
+  return (
+    <FormSection title={t('devconsole~Advanced options')}>
+      <RouteSection route={values.route} resources={values.resources} />
+      <div>
+        <Trans ns="devconsole">
+          Click on the names to access advanced options for{' '}
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/advanced/AdvancedSection.tsx:76]: When using this string with the space included, the i18n tools export can export the same string for all three cases.

```suggestion
          {'Click on the names to access advanced options for '}
```

File: frontend/packages/dev-console/src/components/buildconfig/BuildConfigFormEditor.tsx [hunk 2]
```diff
@@ -13,19 +13,40 @@ import SecretsSection from './sections/SecretsSection';
 import SourceSection from './sections/SourceSection';
 import TriggersSection from './sections/TriggersSection';
 
-export type BuildConfigFormEditorProps = {
+type BuildConfigFormEditorProps = {
   namespace: string;
 };
 
-const BuildConfigFormEditor: React.FC<BuildConfigFormEditorProps> = ({ namespace }) => {
+const List: React.FC<{ namespace: string }> = ({ namespace }) => {
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/buildconfig/BuildConfigFormEditor.tsx:20]: In both other components you reused the parent component props. Not required, but for consistency you can do this here as well

```suggestion
const List: React.FC<BuildConfigFormEditorProps> = ({ namespace }) => {
```

**Review by jerolimov — APPROVED** [review]
Works really well now :+1:  Thanks again for the extra round and cleaning up the components.

/lgtm
/approve

**Pull request merged**